### PR TITLE
initial client-side local evaluation docs updates for js sdk

### PR DESCRIPTION
**Pull request by bgiori**
# Amplitude Developer Docs PR

## Description

Update documentation for client-side local evaluation:
1. Update javascript SDK with new functions and local evaluation info.
2. Update experiment index.md with client-side local evaluation in system overview.
3. Update  local evaluation page with for client-side sdks and implementations.

## Deadline

Today/Tomorrow

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/experiment/general/evaluation/local-evaluation.md [hunk 1]
```diff
@@ -3,18 +3,11 @@ title: Local Evaluation
 description: Detailed information about Amplitude Experiment's local evaluation architecture, limitations, and tradeoffs.
 ---
 
-Server-side local evaluation runs [evaluation logic](./implementation.md) on your server, saving you the overhead incurred by making a network request per user evaluation. The [sub-millisecond evaluation](../performance-and-caching.md#local-evaluation) is perfect for latency-minded systems which need to be performant at scale.
+Local evaluation runs [evaluation logic](./implementation.md) on in the SDK, saving you the overhead incurred by making a network request per user evaluation. The [sub-millisecond evaluation](../performance-and-caching.md#local-evaluation) is perfect for latency-minded systems which need to be performant at scale.
```
Comment (markzegarelli) [inline docs/experiment/general/evaluation/local-evaluation.md:6]:
```suggestion
Local evaluation runs [evaluation logic](./implementation.md) in the SDK, saving you the overhead incurred by making a network request per user evaluation. The [sub-millisecond evaluation](../performance-and-caching.md#local-evaluation) is perfect for latency-minded systems which need to be performant at scale.
```

File: docs/experiment/general/evaluation/local-evaluation.md [hunk 2]
```diff
@@ -3,18 +3,11 @@ title: Local Evaluation
...
-
-    To more easily track exposures on the client-side, [bootstrap](../../sdks/javascript-sdk.md#bootstrapping) the client-side SDK with the variants evaluated server-side and utilize [automatic exposure tracking](../exposure-tracking.md#automatic-exposure-tracking) using one of the [analytics SDK integrations](../../sdks/javascript-sdk.md#integrations).
-
-## Targeting Capabilities
-
-Because local evaluation happens outside of Amplitude, advanced targeting and identity resolution powered by Amplitude Analytics isn't supported. That said, local evaluation allows you to perform consistent bucketing with target segments, which is often sufficient.
+Because local evaluation happens outside of Amplitude, advanced targeting and identity resolution powered by Amplitude Analytics isn't supported. That said, local evaluation allows you to perform consistent bucketing with target segments, which is often enough.
```
Comment (markzegarelli) [inline docs/experiment/general/evaluation/local-evaluation.md:10]:
```suggestion
Local evaluation happens outside of Amplitude, which means advanced targeting and identity resolution powered by Amplitude Analytics isn't supported. That said, local evaluation allows you to perform consistent bucketing with target segments, which is often enough.
```

File: docs/experiment/general/evaluation/local-evaluation.md [hunk 3]
```diff
@@ -31,22 +24,42 @@ Local evaluation is just [evaluation](./implementation.md)--a function which tak
 
 ![Diagram of a local evaluation SDK.](../../../assets/images/experiment/local-evaluation.drawio.svg)
 
-The only non-local part of local evaluation is getting flag configurations from Amplitude Experiment, but this can happen at an interval, and flags can be cached in-memory on the server-side for zero latency access.
+The only non-local part of local evaluation is getting flag configurations from Amplitude Experiment, but this can happen at an interval, and flags can be cached in-memory in the SDK for zero latency access.
```
Comment (markzegarelli) [inline docs/experiment/general/evaluation/local-evaluation.md:27]:
```suggestion
Getting flag configurations from Amplitude Experiment happens on the server. You can configure this to happen at an interval, and Amplitude can cache flags in memory in the SDK for zero-latency access
```

@bgiori Trying to rewrite this in active voice. Is this still accurate? Mainly, is the flag retrieval interval user-configurable?

File: docs/experiment/general/evaluation/local-evaluation.md [hunk 4]
```diff
@@ -31,22 +24,42 @@ Local evaluation is just [evaluation](./implementation.md)--a function which tak
...
     The local evaluation Node.js SDK can be run in edge worker/functions which support JavaScript and a distributed store. Contact your representative or email [experiment@amplitude.com](mailto:experiment@amplitude.com) to learn more.
 
+### Exposure and assignment tracking
+
+Local evaluation SDKs track evaluations differently on the client-side vs on the server-side.
+
+- Client-side SDKs track an [**exposure event**](../exposure-tracking.md) when the user is evaluated due to a variant being accessed from the SDK.
+- Server-side SDKs track an **assignment event** (if configured to do so) when a user is evaluated.
+
+Thus, server-side local evaluation experiments often set the Assignment event as a heuristic for Exposure.
```
Comment (markzegarelli) [inline docs/experiment/general/evaluation/local-evaluation.md:39]:
```suggestion
Server-side local evaluation experiments often set the Assignment event as a heuristic for Exposure.
```

File: docs/experiment/general/evaluation/local-evaluation.md [hunk 5]
```diff
@@ -31,22 +24,42 @@ Local evaluation is just [evaluation](./implementation.md)--a function which tak
...
+
 ## SDKs
 
-Local evaluation is only supported by server-side SDKs which have local evaluation implemented.
+Local evaluation is supported by all server-side SDKs and some client-side SDKs which have local evaluation implemented.
```
Comment (markzegarelli) [inline docs/experiment/general/evaluation/local-evaluation.md:43]:
```suggestion
All server-side SDKs and some client-side (which have local evaluation enabled) support local evaluation.
```

File: docs/experiment/index.md [hunk 6]
```diff
@@ -71,7 +71,7 @@ That said, you can generally split experimentation and feature-flagging systems
 
 !!!done "Easiest way to get started delivering flag and experimenting on your client side application."
 
-Client-side experimentation and feature-flagging involves the client making a request to fetch flags and experiments from Amplitude's [remote evaluation](./general/evaluation/remote-evaluation.md) servers when the application is initialized.
+Client-side implementations assume a single user context. Flag and experiment configurations are loaded on startup for use with [local evaluation](./general/evaluation/local-evaluation.md), and a [remote evaluation](./general/evaluation/remote-evaluation.md) variants are fetched dynamically from Amplitude's remote evaluation service. When a variant is accessed from the SDK, an [exposure event](./general/exposure-tracking.md) is tracked for the user.
```
Comment (markzegarelli) [inline docs/experiment/index.md:74]:
```suggestion
Client-side implementations assume a single user context. Amplitude Experiment loads flag and experiment configurations on startup for use with [local evaluation](./general/evaluation/local-evaluation.md), and  fetches [remote evaluation](./general/evaluation/remote-evaluation.md) variants dynamically from Amplitude's remote evaluation service. When a variant is accessed from the SDK, an [exposure event](./general/exposure-tracking.md) is tracked for the user.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

File: docs/experiment/general/evaluation/local-evaluation.md (hunk 3 above)
Comment (bgiori) [reply docs/experiment/general/evaluation/local-evaluation.md:31]: Ah this sentence is a mess.

```suggestion
Local evaluation flag configurations are loaded from the server on startup and stored in memory for access prior to each evaluation.
```

**Review by markzegarelli — APPROVED** [review]

**Review by tyiuhc — APPROVED** [review]

**Pull request merged**